rfcs/20220128-f64-data_type/Readme.md

```diff
@@ -0,0 +1,72 @@
+# RFC: f64 convolution support
+
+## Introduction & Motivation
+
+Some deep learning applications require higher numerical accuracy than traditionally 
+supported 32-bit floating points, especially for training phase. This RFC addresses
+this requirement by introducing 64-bit, double precision float convolution support in oneDNN.
+We propose adding f64 data type support for convolution on the GPU. Since `reorder` is used to support convolution with different tensor formats, fp64 support will be added to this primitive as well. Other primitives, or engines are not impacted at this point; although library infrastrcture is modified to recognize double-precision floating-point data type.
```

Comment:
Right; but since we are starting with reference ocl kernel, shouldn't it emulate double math on hardware that doesn't have native support? 
I can make a note in terms of performance, this is mostly targeted for hardware with double HW support.

Reply:
@echeresh Do you know if it is possible to support fp64 using OpenCL/DPC++? Do these languages support f64 emulation?

Reply:
I'm not sure emulation in this case is doable/worth it. In OpenCL it's `cl_khr_fp64` extension - we have to enable it before using `double` data type. However if there is no native support in HW (e.g. DG1), this results in a compilation error - OpenCL won't do anything if there is no `double` in hardware.

Reply:
Yes, `cl_khr_fp64` is enabled in my PR #2625 ; I have tested double conv on my laptop's GPU (UHD Graphics) without any compile /runtime errors.

Reply:
@hkianine, f64 is not supported on Gen11, DG1, and DG2. Your laptop is likely a Gen9.